src/rcv/septentrio.c

```diff
@@ -34,55 +36,271 @@
+#define ID_IPSTATUS     4058    /* SBF message id: IP address, gateway and MAC address of Ethernet interface */
+#define ID_WIFIAPSTATUS 4054    /* SBF message id: WiFi status in access point mode */
+#define ID_WIFICLTSTATUS 4096    /* SBF message id: WiFi status in client mode */
+#define ID_DYNDNSSTATUS 4105    /* SBF message id: DynDNS status */
+#define ID_POWERSTATUS  4101    /* SBF message id: Power supply source and voltage */
+#define ID_QUALIND      4082    /* SBF message id: Quality indicators */
+#define ID_DISKSTATUS   4059    /* SBF message id: Internal logging status */
+#define ID_LOGSTATUS    4102    /* SBF message id: Log sessions status  */
+#define ID_RFSTATUS     4092    /* SBF message id: Radio-frequency interference mitigation status  */
+#define ID_P2PPSTATUS   4238    /* SBF message id: P2PP client/server status */
+#define ID_COSMOSTATUS  4243    /* SBF message id: Cosmos receiver service status */
+#define ID_GALAUTHSTATUS 4245   /* SBF message id: Galileo OSNMA authentication status */
+
+/* Miscellaneous Blocks */
+#define ID_RXSETUP      5902     /* SBF message id: General information about the receiver installation*/
+#define ID_RFCOMPS      4084     /* SBF message id: Information on various receiver components  */
+#define ID_RXMESSAGE    4103     /* SBF message id: Receiver Messages */
+#define ID_COMMANDS     4015     /* SBF message id: Commands entered by the user */
+#define ID_COMMENT      5936     /* SBF message id: Comment entered by the user */
+#define ID_BBSMPS       4040     /* SBF message id: Baseband samples */
+#define ID_ASCIIIN      4075     /* SBF message id: ASCII input from external sensor  */
+#define ID_ENCAPSOUT    4097     /* SBF message id: SBF encapsulation of non-SBF messages */
+
+/* Advanced Blocks */
+#define ID_SYSINFO      6000     /* SBF message id: System parameters for maintenance and support  */
+
 
 /* get fields (little-endian) ------------------------------------------------*/
-#define U1(p) (*((uint8_t *)(p)))
-#define I1(p) (*((int8_t  *)(p)))
-static uint16_t U2(uint8_t *p) {uint16_t a; memcpy(&a,p,2); return a;}
-static uint32_t U4(uint8_t *p) {uint32_t a; memcpy(&a,p,4); return a;}
-static int32_t  I4(uint8_t *p) {int32_t  a; memcpy(&a,p,4); return a;}
+#define U1(p) (*((unsigned char *)(p)))
+#define I1(p) (*((signed char *)(p)))
+static uint16_t       U2(unsigned char *p) {uint16_t       u; memcpy(&u,p,2); return u;}
+static uint32_t       U4(unsigned char *p) {uint32_t       u; memcpy(&u,p,4); return u;}
+static uint64_t       U8(unsigned char *p) {uint64_t       u; memcpy(&u,p,8); return u;}
+static float          R4(unsigned char *p) {float          r; memcpy(&r,p,4); return r;}
+static double         R8(unsigned char *p) {double         r; memcpy(&r,p,8); return r;}
+static int32_t        I4(unsigned char *p) {int32_t        u; memcpy(&u,p,4); return u;}
+static int16_t        I2(unsigned char *p) {int16_t        i; memcpy(&i,p,2); return i;}
```

Comment:
This are reading from `raw->buff[]` which is a uint8_t * so might it be cleaner to keep these are uint8_t *? There might be a few exceptions.

Also these could be declared as `const uint8_t *`.

Fwiw I have been converting many of these accessors locally to accept the 'const raw_t *' and the index, and then these can be functions that check the bounds because raw_t has a static buff[] size and also a buffer length - but I understand this level of hardening it not generally wanted.

Reply:
Thanks for the suggestion.